File: Assemblies/Protean.Tools/Integration/Twitter/TwitterVB2/Objects/TwitterTrendLocation.vb
Namespace Integration.Twitter.TwitterVB2

    Public Class TwitterTrendLocation
        Private _ID As String = String.Empty
        Private _LocationName As String = String.Empty
        Private _PlaceTypeCode As String = String.Empty
        Private _PlaceTypeName As String = String.Empty
        Private _CountryName As String = String.Empty
        Private _CountryCode As String = String.Empty
        Private _URL As String = String.Empty


        Public Property ID() As String
            Get
                Return _ID
            End Get
            Set(ByVal value As String)
                _ID = value
            End Set
        End Property


        Public Property LocationName() As String
            Get
                Return _LocationName
            End Get
            Set(ByVal value As String)
                _LocationName = value
            End Set
        End Property


        Public Property PlaceTypeCode() As String
            Get
                Return _PlaceTypeCode
            End Get
            Set(ByVal value As String)
                _PlaceTypeCode = value
            End Set
        End Property


        Public Property PlaceTypeName() As String
            Get
                Return _PlaceTypeName
            End Get
            Set(ByVal value As String)
                _PlaceTypeName = value
            End Set
        End Property


        Public Property CountryName() As String
            Get
                Return _CountryName
            End Get
            Set(ByVal value As String)
                _CountryName = value
            End Set
        End Property


        Public Property CountryCode() As String
            Get
                Return _CountryCode
            End Get
            Set(ByVal value As String)
                _CountryCode = value
            End Set
        End Property


        Public Property URL() As String
            Get
                Return _URL
            End Get
            Set(ByVal value As String)
                _URL = value
            End Set
        End Property


        Public Sub New(ByVal SearchResultNode As System.Xml.XmlNode)
            Me._ID = SearchResultNode("woeid").InnerText

            If SearchResultNode("name") IsNot Nothing Then
                Me._LocationName = SearchResultNode("name").InnerText
            End If

            If SearchResultNode("placeTypeName") IsNot Nothing Then
                Me._PlaceTypeName = SearchResultNode("placeTypeName").InnerText
            End If

            Dim PTNList As System.Xml.XmlNodeList = SearchResultNode.SelectNodes("placeTypeName")

            If PTNList(0).Attributes("code") IsNot Nothing Then
                Me._PlaceTypeCode = PTNList(0).Attributes("code").Value
            End If

            If SearchResultNode("country") IsNot Nothing Then
                Me._CountryName = SearchResultNode("country").InnerText
            End If

            Dim CountryList As System.Xml.XmlNodeList = SearchResultNode.SelectNodes("country")

            If CountryList(0).Attributes("code") IsNot Nothing Then
                Me._CountryCode = CountryList(0).Attributes("code").Value
            End If

            If SearchResultNode("url") IsNot Nothing Then
                Me._URL = SearchResultNode("url").InnerText
            End If
        End Sub
    End Class
End Namespace

File: Assemblies/Protean.Tools_Net7/Integration/Twitter/TwitterVB2/Objects/TwitterTrendLocation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Protean.Tools.Integration.Twitter.TwitterVB2.Objects
{
    public partial class TwitterTrendLocation
    {
        private string _ID = string.Empty;
        private string _LocationName = string.Empty;
        private string _PlaceTypeCode = string.Empty;
        private string _PlaceTypeName = string.Empty;
        private string _CountryName = string.Empty;
        private string _CountryCode = string.Empty;
        private string _URL = string.Empty;


        public string ID
        {
            get
            {
                return _ID;
            }
            set
            {
                _ID = value;
            }
        }


        public string LocationName
        {
            get
            {
                return _LocationName;
            }
            set
            {
                _LocationName = value;
            }
        }


        public string PlaceTypeCode
        {
            get
            {
                return _PlaceTypeCode;
            }
            set
            {
                _PlaceTypeCode = value;
            }
        }


        public string PlaceTypeName
        {
            get
            {
                return _PlaceTypeName;
            }
            set
            {
                _PlaceTypeName = value;
            }
        }


        public string CountryName
        {
            get
            {
                return _CountryName;
            }
            set
            {
                _CountryName = value;
            }
        }


        public string CountryCode
        {
            get
            {
                return _CountryCode;
            }
            set
            {
                _CountryCode = value;
            }
        }


        public string URL
        {
            get
            {
                return _URL;
            }
            set
            {
                _URL = value;
            }
        }


        public TwitterTrendLocation(System.Xml.XmlNode SearchResultNode)
        {
            _ID = SearchResultNode["woeid"].InnerText;

            if (SearchResultNode["name"] != null)
            {
                _LocationName = SearchResultNode["name"].InnerText;
            }

            if (SearchResultNode["placeTypeName"] != null)
            {
                _PlaceTypeName = SearchResultNode["placeTypeName"].InnerText;
            }

            var PTNList = SearchResultNode.SelectNodes("placeTypeName");

            if (PTNList[0].Attributes["code"] != null)
            {
                _PlaceTypeCode = PTNList[0].Attributes["code"].Value;
            }

            if (SearchResultNode["country"] != null)
            {
                _CountryName = SearchResultNode["country"].InnerText;
            }

            var CountryList = SearchResultNode.SelectNodes("country");

            if (CountryList[0].Attributes["code"] != null)
            {
                _CountryCode = CountryList[0].Attributes["code"].Value;
            }

            if (SearchResultNode["url"] != null)
            {
                _URL = SearchResultNode["url"].InnerText;
            }
        }
    }
}
